File: apps/android-usage/companion/src/test/kotlin/com/healthmes/companion/Fixtures.kt
package com.healthmes.companion

/** Loads contract fixtures from src/test/resources. */
object Fixtures {

    fun load(name: String): String =
        checkNotNull(javaClass.classLoader?.getResource(name)) {
            "missing test fixture: $name"
        }.readText()

    /** Populated payload matching the /v1/briefing/glance contract verbatim. */
    fun full(): String = load("glance_full.json")

    /** Empty-database shape (all-null energy, no blocks/alerts/decision). */
    fun empty(): String = load("glance_empty.json")
}

File: apps/windows-companion/tests/HealthMes.Glance.Core.Tests/Fixtures.cs
namespace HealthMes.Glance.Core.Tests;

/// <summary>Loads the pinned contract fixtures copied next to the test binary.</summary>
public static class Fixtures
{
    public static string Read(string name) =>
        File.ReadAllText(Path.Combine(AppContext.BaseDirectory, "Fixtures", name));

    /// <summary>iOS reference payload (apps/ios-companion/Tests/Fixtures/glance.json).</summary>
    public static string GlanceIos() => Read("glance.json");

    /// <summary>Android "full" payload (companion/src/test/resources/glance_full.json).</summary>
    public static string GlanceFull() => Read("glance_full.json");

    /// <summary>Android "honest empty" payload (all-null energy, no blocks/alerts).</summary>
    public static string GlanceEmpty() => Read("glance_empty.json");

    public static string AlertsPage() => Read("alerts_page.json");

    public static string WeeklyReport() => Read("weekly_report.json");
}
